docs/backend/guides/open-telemetry.md

```diff
@@ -0,0 +1,251 @@
+
+### Go package
+
+The main initialization code for registering telemetry can be found in `pkg/telemetry/init.go`
+
+The custom code for collecting database connection stats also lives within the package at `pkg/telemetry/db.go`.
+
+The code for hooking into the Go runtime library for memory stats lives within the package at `pkg/telemetry/runtime.go`
+
+### Instrumentation
+
+The ecosystem of Open Telemetry comes with many built in and 3rd party libraries for plugging into our existing code with little or no customization.
+
+#### Routing
+
+Milmove uses the Gorilla mux router so we can use the existing [otelmux library](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/github.com/gorilla/mux/otelmux) middleware for our non-swagger routes.
+
+```golang
+router.Use(otelmux.Middleware("auth"))
+```
+
+#### HTTP handler
+
+Similar to the otelmux library the [otelhttp library](https://github.com/trussworks/otelhttp) is designed to wrap HTTP handler functions. It has built in events (read & write) to report the request size, response size, and duration of the http request.  It is more configurable than the otelmux middleware allowing you to filter out certain routes and other controls.
+
+```golang
+otelhttp.NewHandler(router, "server-tls", []otelHTTP.Option{otelHTTP.ReadEvents, otelHTTP.WriteEvents})
+```
+
+Note: This implementation has been forked to the Trussworks GitHub to address bug fixes.
+
+#### SQL 
+
+The [XSAM/otelsql](https://github.com/XSAM/otelsql) library wraps the queries to our Postgres database and appends a span to the trace with the statement and duration information.  Query parameters may be hidden to not log sensitive information.  It can also be used to log calls to cursor rows when paging through larger result sets.
+
+#### Go
+
+The [Go runtime instrumentation library](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/runtime) primarily provides statistics on memory usage and garbage collection cycles that may degrade performance.
+
+## Data collection
+
+Data from the Milmove app is sent to the Open Telemetry collector, which processes the trace, metric, and log data and exports it to the services of our choice.  In AWS the collector exports to the Cloudwatch and AWS X-Ray services for storage and analysis.  When run locally the data is exported to the Elastic APM Server running Elasticsearch and Kibana. 
+
+### Traces
+
+*Common Trace fields*
+- ID
+- Parent ID
+- Trace ID
+- Name
+- Kind
+- Start time
+- End time
+- Status code
+- Status message
+
+#### HTTP Requests
+
+```
+http.client_ip
+http.flavor
+http.host
+http.method
+http.scheme
+http.server_name
+http.status_code
+http.target
+http.user_agent
+http.wrote_bytes
+net.host_ip
+net.host_port
+net.peer_ip
+net.peer_port
+net.transport
+```
+
+#### Database SQL Statements
+
```

Comment:
What're these used for?

Reply:
`sql.conn.query` - This is the more useful span that will be added to the trace request that logs the sql query and the duration so we could identify long running database queries.

`sql.rows` - I don't see much data here besides the duration. My guess is that this reports when you return a result set of records from a query as a cursor then loop through them like for paginated data.  I don't think we're doing that too many places in the codebase although Pop may be doing it under the hood.

You can see some of the real data in AWS now https://console.amazonaws-us-gov.com/cloudwatch/home?region=us-gov-west-1#servicelens:traces/1-61a67e8c-21e5141bacfa25f710da1a5b?~(query~(filter~(node~'*7e*28name*7e*27sql.conn.exec*7etype*7e*27Database*2a3a*2a3aSQL*7edataSource*7e*27xray*29~groupFilters~()))~context~())

![Screen Shot 2021-11-30 at 4 18 31 PM](https://user-images.githubusercontent.com/52669884/144130569-b70b940c-2b87-4c00-bd20-ad6792db1a45.png)

Reply:
Thanks for this question, I realized I was missing `sql.conn.exec` which are for updates/inserts and sql.conn.query are only selects statements apparently.